geth/node/syncpoll.go

```diff
@@ -0,0 +1,70 @@
+package node
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/ethereum/go-ethereum/eth/downloader"
+	"github.com/ethereum/go-ethereum/les"
+)
+
+// errors
+var (
+	ErrSyncStartTimeout = errors.New("node synchronization timeout before start")
+	ErrSyncAborted      = errors.New("node synchronization timeout before completion")
+)
+
+// SyncPoll provides a structure that allows us to check the status of
+// ethereum node synchronization.
+type SyncPoll struct {
+	downloader *downloader.Downloader
+}
+
+// NewSyncPoll returns a new instance of SyncPoll.
+func NewSyncPoll(leth *les.LightEthereum) *SyncPoll {
+	return &SyncPoll{
+		downloader: leth.Downloader(),
+	}
+}
+
+// Poll returns a channel which allows the user to listen for a done signal
```

Comment:
Perhaps outdated comment here? `Poll` returns error, not channel, right?

Reply:
Yes, outdated indeed, will rewrite. Thanks. :)